x-pack/plugins/security_solution/public/common/components/exceptions/helpers.tsx

```diff
@@ -341,3 +358,159 @@ export const filterExceptionItems = (
     []
   );
 };
+
+export const formatExceptionItemForUpdate = (
+  exceptionItem: ExceptionListItemSchema
+): UpdateExceptionListItemSchema => {
+  const {
+    created_at,
+    created_by,
+    list_id,
+    tie_breaker_id,
+    updated_at,
+    updated_by,
+    ...fieldsToUpdate
+  } = exceptionItem;
+  return {
+    ...fieldsToUpdate,
+  };
+};
+
+export const enrichExceptionItemsWithComments = (
+  exceptionItems: Array<ExceptionListItemSchema | CreateExceptionListItemSchema>,
+  comments: Array<Comments | CreateComments>
+): Array<ExceptionListItemSchema | CreateExceptionListItemSchema> => {
+  return exceptionItems.map((item: ExceptionListItemSchema | CreateExceptionListItemSchema) => {
+    return {
+      ...item,
+      comments,
+    };
+  });
+};
+
+export const enrichExceptionItemsWithOS = (
+  exceptionItems: Array<ExceptionListItemSchema | CreateExceptionListItemSchema>,
+  osTypes: string[]
+): Array<ExceptionListItemSchema | CreateExceptionListItemSchema> => {
+  const osTags = osTypes.map((os) => `os:${os}`);
+  return exceptionItems.map((item: ExceptionListItemSchema | CreateExceptionListItemSchema) => {
+    const newTags = item._tags ? union(item._tags, osTags) : [...osTags];
+    return {
+      ...item,
+      _tags: newTags,
+    };
+  });
+};
+
+export const getMappedNonEcsValue = ({
+  data,
+  fieldName,
+}: {
+  data: TimelineNonEcsData[];
+  fieldName: string;
+}): string[] | undefined => {
+  const item = data.find((d) => d.field === fieldName);
+  if (item != null && item.value != null) {
+    return item.value;
+  }
+  return undefined;
```

Comment:
Could you return an empty array here and then you don't have to do checks like the following and the return type is just `string[]`: `getMappedNonEcsValue({ data: alertData, fieldName: 'file.hash.sha1' }) ?? []`

Reply:
To be addressed in follow up PR